src/chocolatey.resources/helpers/functions/Get-Installed.ps1

```diff
@@ -0,0 +1,46 @@
+﻿## Get-Installed
+##############################################################################################################
+## Returns a list of installed applications.
+## Registry based replacement for "Get-WmiObject -Class Win32_Product".
+##############################################################################################################
+
+function Get-Installed {
+param(
+  [string] $NonMSIOption = 'HideNonMSI' # You may use 'ShowNonMSI' option
+)
+
+  if(($NonMSIOption -ine 'HideNonMSI') -and ($NonMSIOption -ine 'ShowNonMSI')) {
+    throw "Unsupported argument: $NonMSIOption"
+  }

```

Comment:
@justdanpo: If you want, you can use a PowerShell `[switch]` parameter instead. A PowerShell switch is used just like a boolean value in checks. However, the syntax for supplying the parameter is different. For example, suppose you have a switch parameter called `ShowNonMSI` that defaults to `$false`. Here is how you would use it:

Usage|Result
---------|--------
`Get-Installed` | `$ShowNonMSI` is `$false`
`Get-Installed -ShowNonMSI` | `$ShowNonMSI` is `$true`

You would declare the parameter like this:

```powershell
param(
    [switch]$ShowNonMSI = $false
)
```

Does this help?